source/postprocess/tracers.cc

```diff
@@ -0,0 +1,406 @@
+/*
+  Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/global.h>
+#include <aspect/postprocess/tracers.h>
+#include <aspect/simulator_access.h>
+
+#include <aspect/particle/generator/interface.h>
+#include <aspect/particle/integrator/interface.h>
+#include <aspect/particle/interpolator/interface.h>
+#include <aspect/particle/property/interface.h>
+
+#include <boost/archive/text_oarchive.hpp>
+#include <boost/archive/text_iarchive.hpp>
+
+#include <stdio.h>
+#include <unistd.h>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    Tracers<dim>::Tracers ()
+      :
+      initialized(false),
+      next_data_output_time(std::numeric_limits<double>::quiet_NaN())
+    {}
+
+
+    template <int dim>
+    Tracers<dim>::~Tracers ()
+    {
+    }
+
+    template <int dim>
+    void
+    Tracers<dim>::initialize()
+    {
+
+    }
+
+    template <int dim>
+    const Particle::World<dim> &
+    Tracers<dim>::get_particle_world() const
+    {
+      return world;
+    }
+
+    template <int dim>
+    Particle::World<dim> &
+    Tracers<dim>::get_particle_world()
+    {
+      return world;
+    }
+
+    template <int dim>
+    std::pair<std::string,std::string>
+    Tracers<dim>::execute (TableHandler &statistics)
+    {
+      // Let the generator add the specified number of particles if we have not
+      // already done so
+      if (!initialized)
```

Comment:
If you do this in the `execute()` function of a postprocessor, then this means that you don't have particles during most of the first time step. You can't make particles active this way. I think this needs to run during initialization somewhere. Can't you do this in the function that overloads `Postprocess::Interface::initialize()`?